source/Halibut.Tests/WhenCancellingARequestForAPollingTentacle.cs

```diff
@@ -206,50 +187,61 @@ public async Task<ResponseMessage> QueueAndWaitAsync(RequestMessage request, Req
         /// </summary>
         class CancelWhenRequestDequeuedPendingRequestQueueFactory : IPendingRequestQueueFactory
         {
-            readonly CancellationTokenSource[] cancellationTokenSources;
+            readonly CancellationTokenSource cancellationTokenSource;
+            readonly Func<bool> shouldCancel;
+            readonly Action<ResponseMessage> onResponseApplied;
             readonly IPendingRequestQueueFactory inner;
 
-            public CancelWhenRequestDequeuedPendingRequestQueueFactory(IPendingRequestQueueFactory inner, CancellationTokenSource[] cancellationTokenSources)
+            public CancelWhenRequestDequeuedPendingRequestQueueFactory(IPendingRequestQueueFactory inner, CancellationTokenSource cancellationTokenSource, Func<bool> shouldCancel, Action<ResponseMessage> onResponseApplied)
             {
-                this.cancellationTokenSources = cancellationTokenSources;
+                this.cancellationTokenSource = cancellationTokenSource;
+                this.shouldCancel = shouldCancel;
                 this.inner = inner;
-            }
-
-            public CancelWhenRequestDequeuedPendingRequestQueueFactory(IPendingRequestQueueFactory inner, CancellationTokenSource cancellationTokenSource): this(inner, new []{ cancellationTokenSource })
-            {
+                this.onResponseApplied = onResponseApplied;
             }
 
             public IPendingRequestQueue CreateQueue(Uri endpoint)
             {
-                return new Decorator(inner.CreateQueue(endpoint), cancellationTokenSources);
+                return new Decorator(inner.CreateQueue(endpoint), cancellationTokenSource, shouldCancel, onResponseApplied);
```

Comment:
Nit: maybe `shouldCancelOnDequeue`?

Reply:
Done

Reply:
Awesome! 

Although, I noticed the renamed stopped happening from line 209 downwards.

Just a nit 🙂